sound/soc/sof/intel/hda.c

```diff
@@ -1207,21 +1209,24 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 
 			if (ids[j].link_id != link_id ||
 			    ids[j].id.part_id != part_id ||
-			    ids[j].id.mfg_id != mfg_id)
+			    ids[j].id.mfg_id != mfg_id ||
+			    ids[j].id.sdw_version != version)
 				continue;
 
 			/* find out how many identical parts are expected */
 			for (k = 0; k < link->num_adr; k++) {
 				u64 adr2 = link->adr_d[k].adr;
-				unsigned int part_id2, link_id2, mfg_id2;
+				unsigned int part_id2, link_id2, mfg_id2, version2;
 
 				mfg_id2 = SDW_MFG_ID(adr2);
 				part_id2 = SDW_PART_ID(adr2);
 				link_id2 = SDW_DISCO_LINK_ID(adr2);
+				version2 = SDW_VERSION(adr2);
```

Comment:
Is this a theoretical problem or a real one?

I think it's correct to use the version but was there a platform where this problem appeared?

Reply:
> Is this a theoretical problem or a real one?
> 
> I think it's correct to use the version but was there a platform where this problem appeared?

Yes, the ADL-PS RVP uses rt711-sdca codec. The new snd_soc_acpi_link_adr will be similar to adl_rvp[] except the codec is rt711-sdca instead of rt711.

Reply:
ok, thanks for the precision